### ISSUE TITLE
Incorrect encoding causes incorrect output of varchar in Windows

### ISSUE THREAD

**Issue by ElsLommelen** [r-dbi/odbc#371]
### Issue Description and Expected Result
Non ascii symbols saved as a varchar in a SQL Server database are returned incorrectly by `dbFetch()` on a Windows machine.  E.g. the symbol `"µ"` in SQL Server is returned in R as `"\xb5"`.  On the other hand, the output is correct when saved as a nvarchar in SQL Server.  In Linux, symbols are presented correctly in both cases.

The [declared](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/Encoding) encoding of the output is `"UTF-8"` in both Windows and Linux, and both for varchar as nvarchar. For rather common symbols as `"µ"`, a workaround for varchar in Windows is to declare the encoding of the output column active as the native encoding of Windows:
`Encoding(<output_column>) <- "latin1"`
This does however not solve the problem in more complex symbols such as the symbols for male or female (they are always presented as a `"?"`).

On the other hand, when performing the code `Encoding(<output_column>) <- "latin1"` for symbols that were in nvarchar in SQL Server, these symbols (that were presented well in first instance) are presented incorrect (e.g. `"µ"` as `"Âµ"`).

So it seems that the declared encoding of a varchar in the odbc package is not appropriate to be used in Windows.  I hope the above explanation is sufficient to pinpoint and solve this problem?  I don't know how to make a reprex for SQL Server.

The same behaviour as varchar occurs for varchar(max), so please provide the same solution for this data type.
(Similar to nvarchar, data types nvarchar(max) and ntext don't have this problem.)

### Database
SQL Server 2016 queried from a Windows machine

**Comment by shrektan** [comment]
I’m quite confused. Setting the encoding param of dbConnect() to the encoding of the server can’t solve your issue?

**Comment by ElsLommelen** [comment]
Oups, I had never noticed the encoding param in dbConnect().
I tested it, and this gives the same result as the workaround I did: it is good a solution for rather common symbols, so I'll surely start using it.  But less common symbols such as `"\u2640"` (the symbol for female) in varchars still result in a `"?"`.  So these symbols are not correctly presented when originating from a varchar field, but they are correctly presented from a nvarchar field.
